src/cargo/core/source/mod.rs

```diff
@@ -44,6 +44,9 @@ pub trait Source {
     /// Ensure that the source is fully up-to-date for the current session on the next query.
     fn invalidate_cache(&mut self);
 
+    /// If quiet, the source should not display any progress or status messages.
+    fn set_quiet(&mut self, quiet: bool);
```

Comment:
While I think the overall improvement makes this not a blocker, from an API design perspective, it feels weird to me that are individually silencing sources like this.